GitUI/UserControls/RevisionGrid/FilterInfo.cs

```diff
@@ -423,31 +423,27 @@ private void GetBranchRevisionFilter(ArgumentBuilder filter)
                 // All commits
                 filter.Add("--reflog");
             }
-            else if (IsShowCurrentBranchOnlyChecked || (IsShowFilteredBranchesChecked && !string.IsNullOrWhiteSpace(BranchFilter)))
+            else if (IsShowCurrentBranchOnlyChecked && !string.IsNullOrWhiteSpace(currentBranch.Value))
             {
-                // See handling for --all below, explicitly add most recent stash
-                if (AppSettings.ShowStashes)
-                {
-                    // stash@{0} requires that the repo has commits and --glob=refs/stash is handled as refs/stash/*,
-                    // so this weird pattern must be used: "--glob=refs/stas[h]"
-                    filter.Add($"--glob={GitRefName.RefsStashPrefix.Remove(GitRefName.RefsStashPrefix.Length - 1, 1)}[h]");
-                }
+                // Git default, no option by default (stashes is special).
+                // Add as filter anyway as the branch must exist and the repo must contain commits
+                // otherwise Git will exit with errors.
 
-                if (IsShowCurrentBranchOnlyChecked)
-                {
-                    // Default git-log (no option), only current branch (HEAD)
-                    if (AppSettings.ShowStashes)
-                    {
-                        filter.Add("HEAD");
-                    }
-                }
-                else
-                {
-                    // Show filtered branches only
-                    // Do not add suffix /* to the filter
-                    bool isSimpleBranchFilter = BranchFilter.IndexOfAny(new[] { '?', '*', '[' }) == -1;
-                    filter.Add(isSimpleBranchFilter ? BranchFilter : $"--branches={BranchFilter}");
-                }
+                AddFirstStashRef();
+
+                // current branch must be an option as the string may not exist
+                filter.Add($"--branches={GetFilterRefName(currentBranch.Value)}");
+            }
+            else if (IsShowFilteredBranchesChecked && !string.IsNullOrWhiteSpace(BranchFilter))
+            {
+                // Show filtered branches only
+
+                AddFirstStashRef();
+
+                // Simple branch filter without wildcards
+                // Git adds the suffix "/*" if there is no wildcard
+                bool isSimpleBranchFilter = BranchFilter.IndexOfAny(new[] { '?', '*', '[' }) == -1;
+                filter.Add(isSimpleBranchFilter ? BranchFilter : $"--branches={BranchFilter}");
```

Comment:
[nit] Please don't wrap lines too short, 120-150 range is generally something worth striving for
``suggestion
                // If BranchFilter contains wildcards, "--branches=" must be prepended.
                // But use the simple branch filter if without wildcards in order to avoid git adding implicit /* to the "--branches=" filter then.
```

Reply:
> Please don't wrap lines too short, 120-150 range is generally something worth striving for

Avoid over 80 chars if possible, really exceptions if more than 120 chars. It is then possible to have several editors open side by side. Especially for Visual Studio, not great with multiple monitors.

Reworded slightly. This will likely be changed anyway, if a similar (but different) check is done for filtered branches.
The second line would have been 163 characters

                // If BranchFilter contains wildcards, "--branches=" must be prepended.
                // Use the simple branch filter if without wildcards (must be Git reference)
                // in order to avoid git adding implicit /* to the "--branches=" filter.

Reply:
80?! Developers are generally constrained vertically rather than horizontally.

This is how 80 looks for me:
![image](https://user-images.githubusercontent.com/4403806/226214685-5c97a975-442b-4b26-85e8-449bb3f0df99.png)

Reply:
Use several editors if you do not want to use move to other view in npp.
You get much more code on your screen with that.
(I use VSC or npp at work, only with GE I use VS)

VS has some OH, but two instances are OK
![image](https://user-images.githubusercontent.com/6248932/226215147-a1528918-6a05-4786-bc9a-ac013f34c2d7.png)